src/nupic/algorithms/Anomaly.cpp

```diff
@@ -29,33 +29,49 @@
 #include "nupic/algorithms/Anomaly.hpp"
 #include "nupic/utils/Log.hpp"
 #include "nupic/utils/MovingAverage.hpp"
+#include "nupic/types/SdrTools.hpp" // sdr::Intersection
 
 using namespace std;
+using namespace nupic;
+using namespace nupic::algorithms::anomaly;
+using namespace nupic::sdr;
 
 namespace nupic {
 namespace algorithms {
 namespace anomaly {
+Real32 computeRawAnomalyScore(SDR& active,
+                              SDR& predicted) {
 
-
-Real computeRawAnomalyScore(const vector<UInt>& active,
-                              const vector<UInt>& predicted)
-{
   // Return 0 if no active columns are present
-  if (active.size() == 0) {
+  if (active.getSum() == 0) {
     return 0.0f;
   }
 
-  set<UInt> active_{active.begin(), active.end()};
-  set<UInt> predicted_{predicted.begin(), predicted.end()};
-  vector<UInt> predictedActiveCols;
+  NTA_CHECK(active.dimensions == predicted.dimensions);
 
   // Calculate and return percent of active columns that were not predicted.
-  set_intersection(active_.begin(), active_.end(), predicted_.begin(),
-                   predicted_.end(), back_inserter(predictedActiveCols));
+  Intersection both(active, predicted); //TODO allow Intersection to act on const SDR, so active,pred can be const
```

Comment:
Am I doing something wrong with the intersect. computation? 

code below 
```
cout << "act " << active << endl;
  cout << "pred " << predicted << endl;
  cout << "x " << both;
```

gives me 
act 3,5,7
pred 3, 5, 7
BUT x (3) <and empty here> 

This is from `AnomalyTest PerfectMatch`, I checked the sdr::Intersect usage from its unit-test example.